### PR TITLE
Backend testing

### PR DESCRIPTION
**Pull request by thomas-galligan**

### CODE REVIEW

File: server/handlers/getSurvey.js [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 const getSurvey = async (req, res) => {
   try {
     // checks if params contains key id or if params itself is just one string
-    const id = !req.params.id ? req.params : req.params.id;
-
+    const { id } = req.params
     const result = await readSurvey(id.toString());
     res.status(200).json(result);
   } catch (err) {
+    console.log(err)
```
Comment (AntonioGargaro) [inline server/handlers/getSurvey.js]: Should be `console.warn` but ill let you off

File: server/queries/updateSurvey.test.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
 /* eslint-disable no-undef */
-const { initDb, closeDb } = require('../databaseConnection');
 // const { ObjectID } = require('mongodb');
```
Comment (AntonioGargaro) [inline server/queries/updateSurvey.test.js]: don't need comment

File: server/handlers/getSurvey.js [hunk 3]
```diff
@@ -1,13 +1,15 @@
+
+/* eslint-ignore no-console */
 const readSurvey = require('../queries/readSurvey');
 
 const getSurvey = async (req, res) => {
   try {
     // checks if params contains key id or if params itself is just one string
-    const id = !req.params.id ? req.params : req.params.id;
-
+    const { id } = req.params
```
Comment (marthaklambert) [inline server/handlers/getSurvey.js]: i know this looks weird but im pretty sure it was the only way of fixing @Joaoviana bug with sending params. I would ask him before removing this

File: server/queries/updateSurvey.test.js [hunk 4]
```diff
@@ -12,16 +12,18 @@ afterEach(() => {
...
-it('Testing updateSurvey updates a document successfully.', async (done) => {
-  done();
+it('Testing updateSurvey updates a document successfully.', async () => {
   // const originalDocument = {
-  //   id: ObjectID('507f1f77bcf86cd799439001'),
+  //   _id: ObjectID('508f1f99bcf86cd799439014'),
   //   title: 'Graduates Onboarding',
   //   description:
   //     'A survey description that is about 1 sentence long yeah blah blah',
   //   status: 'draft',
```
Comment (AntonioGargaro) [inline server/queries/updateSurvey.test.js]: Dont need all these comments

File: server/queries/updateSurvey.test.js (hunk 2 above)
Comment (thomas-galligan) [reply server/queries/updateSurvey.test.js]: done

File: server/queries/updateSurvey.test.js (hunk 4 above)
Comment (thomas-galligan) [reply server/queries/updateSurvey.test.js]: these have been removed now

File: server/handlers/getSurvey.test.js [hunk 5]
```diff
@@ -1,31 +1,20 @@
...
+  it('Responds with JSON and status 200 when the survey has been found successfully.', async () => {
+
+    await initDb();
+
+    const res = await request(app)
+      .get('/surveys/508f1f99bcf86cd799439014')
+      .set('Accept', 'application/json')
+      .expect('Content-Type', /json/)
+
+    expect(res.status).toEqual(200)
```
Comment (TKostrzewski) [inline server/handlers/getSurvey.test.js]: Can we test the result being returned? If the status is 200, I expect something is sent back, could we make an assertion on this please?

File: server/handlers/getSurvey.test.js [hunk 6]
```diff
@@ -1,31 +1,20 @@
...
-    //   .end((err, res) => {
-    //     if (err) {
-    //       return done(err);
-    //     }
-    //     expect(res.body.auth).toEqual(true);
-    //     return done();
-    //   });
+  it('Responds with JSON and status 200 when the survey has been found successfully.', async () => {
+
+    await initDb();
```
Comment (TKostrzewski) [inline server/handlers/getSurvey.test.js]: In some tests, we are calling initDb() and ending with closeDb(), in other tests, we are using beforeEach and afterEach. Not important as it works but something to consider for consistency.

File: server/queries/createSurvey.test.js [hunk 7]
```diff
@@ -1,45 +1,45 @@
...
+    ],
+  };
 
-  // const dummyQuestionsArray = [{ questionId: '4321211412', position: 2 }];
+  const dummyQuestionsArray = [{ questionId: '4321211412', position: 2 }];
 
-  // const result = await createSurvey(dummySurvey, dummyQuestionsArray);
-  // expect(result).toEqual('success');
+  const result = await createSurvey(dummySurvey, dummyQuestionsArray);
+  expect(result).toEqual('success');
```
Comment (TKostrzewski) [inline server/queries/createSurvey.test.js]: Can we make an assertion on the status code please?

File: server/queries/readSurvey.test.js [hunk 8]
```diff
@@ -1,66 +1,95 @@
...
+          'agree',
+          'strongly agree',
+        ],
+      },
+    ],
+    responses: [],
+  };
+  expect.assertions(1);
+  const readDocument = await readSurvey('508f1f99bcf86cd799439014');
+  expect(readDocument).toStrictEqual(documentToFind);
```
Comment (TKostrzewski) [inline server/queries/readSurvey.test.js]: can we make an assertion on the status code please? Not major here as this assertion looks pretty solid.

File: server/handlers/getSurvey.js (hunk 3 above)
Comment (thomas-galligan) [reply server/handlers/getSurvey.js]: if we're writing hacky stuff like this then it points to a deeper problem with our code, I'd rather we left it like this and if we encounter problems then we can address them

File: server/queries/updateSurvey.test.js [hunk 9]
```diff
@@ -12,65 +12,13 @@ afterEach(() => {
...
-  // expect(updatedDocument.dateClosed).toEqual(2) &&
-  //   expect(originalDocument.dateClosed).toStrictEqual('');
+it('Testing updateSurvey updates a document successfully.', async () => {
+
+  const queryResult = await updateSurvey('508f1f99bcf86cd799439014', {
+    status: 'published',
+    datePublished: 1,
+    dateClosed: 2,
+  });
+  expect(queryResult.result.nModified).toEqual(1);
```
Comment (TKostrzewski) [inline server/queries/updateSurvey.test.js]: nModified? Not sure what that is

Comment (thomas-galligan) [reply server/queries/updateSurvey.test.js]: number of documents modified, it's just a property that mongo returns

File: server/handlers/getSurvey.js (hunk 3 above)
Comment (Joaoviana) [reply server/handlers/getSurvey.js]: cool, fine with that!

File: server/queries/readSurvey.test.js (hunk 8 above)
Comment (thomas-galligan) [reply server/queries/readSurvey.test.js]: will do in a future iteration, just wanted to get some tests in for now

File: server/queries/createSurvey.test.js (hunk 7 above)
Comment (thomas-galligan) [reply server/queries/createSurvey.test.js]: will do in the next iteration, this is more of an MVP test for now

File: server/handlers/getSurvey.test.js (hunk 6 above)
Comment (thomas-galligan) [reply server/handlers/getSurvey.test.js]: agreed there is some inconsistency, but given time pressures I'm not going to comb through on this level of granularity. Thanks for raising this though!

File: server/handlers/getSurvey.js (hunk 1 above)
Comment (thomas-galligan) [reply server/handlers/getSurvey.js]: thank you kind sir

File: server/handlers/getSurvey.test.js (hunk 5 above)
Comment (thomas-galligan) [reply server/handlers/getSurvey.test.js]: Had some issues getting that working so it's removed for now. Will do in the next iteration

**Pull request merged**